### DISCUSSION TITLE
dagger.gql lock file

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#12067] [Ideas]
There should be a way to repeatedly run dagger pipelines with a set of extensions.

When extensions in dagger.gql are first pulled via git refs, we should record the sha in a dagger.lock file.

**Comment by sipsma** [comment]
Agreed 100%

We should see if we can re-use the new related feature coming in buildkit soon: https://github.com/moby/buildkit/pull/2943

**Comment by jedevc** [comment]
x-linking https://github.com/dagger/dagger/pull/10975
